pandas/core/indexes/base.py

```diff
@@ -2404,6 +2408,28 @@ def is_all_dates(self) -> bool:
         )
         return self._is_all_dates
 
+    @final
+    @classmethod
+    def _is_num_index(cls) -> bool:
```

Comment:
what is the ultimate purpose of this? it is very odd to have this type of method

Reply:
For backward compat, we need to know when something is a `NumericIndex` but not a `Int64Index`, a `UInt64Index` or `Float64Index`. This does that.

Reply:
I've added an atrribute  boolean `_is_numeric_inde` instead. That simolifies this a bit.

We need to know when something is a `NumericIndex` but not a `Int64Index`, a `UInt64Index` or `Float64Index`, so we need something like tht until pandas 2.0/when we drop `Int64Index` etc.